streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -1142,7 +1147,7 @@ void retryWithBackoff(final Map<TaskId, Set<TopicPartition>> tasksToBeCreated) {
                     }
                 }
 
-                if (tasksToBeCreated.isEmpty()) {
+                if (tasksToBeCreated.isEmpty() || backoffTimeMs > rebalanceTimeoutMs) {
```

Comment:
`MAX_POLL_INTERVAL_MS_CONFIG` is absolute max time till this thread is kicked out. So this can be the upper limit for `backoffTimeMs`. I am not sure how can we track total time this thread was re balancing.  Also not throwing an exception but carrying on is OK, because it would try to commit offsets and it will fail that time, because it is no longer the owner of these partitions and already kicked out of the group. So it will try to join again and get new partitions.